src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -677,43 +681,49 @@ public function getEditShipmentForm(int $orderId, Request $request): Response
 
-        $form = $this->createForm(MergeShipmentType::class, null, $formData);
-        $form->handleRequest($request);
+            $form = $this->createForm(MergeShipmentType::class, null, $formData);
+            $form->handleRequest($request);
 
-        if (!$form->isSubmitted() || !$form->isValid()) {
-            $this->addFlash('error', 'Invalid merge shipment form.');
+            if (!$form->isSubmitted() || !$form->isValid()) {
+                $this->addFlash('error', 'Invalid merge shipment form.');
 
-            return $this->redirectToRoute('admin_orders_view', ['orderId' => $orderId]);
-        }
+                return $this->redirectToRoute('admin_orders_view', ['orderId' => $orderId]);
+            }
 
-        /** @var OrderShipment $targetShipment */
-        $targetShipment = $form->get('merge_to_shipment')->getData();
-        $targetShipmentId = $targetShipment->getId();
+            /** @var OrderShipment $targetShipment */
+            $targetShipment = $form->get('merge_to_shipment')->getData();
+            $targetShipmentId = $targetShipment->getId();
 
-        $submittedData = $request->request->all('merge_shipment');
-        $selectedProducts = [];
+            $submittedData = $request->request->all('merge_shipment');
+            $selectedProducts = [];
 
-        foreach ($submittedData as $key => $value) {
-            if (str_starts_with($key, 'product_') && $value) {
-                $orderDetailId = (int) str_replace('product_', '', $key);
-                $quantityKey = 'quantity_' . $orderDetailId;
+            foreach ($submittedData as $key => $value) {
+                if (str_starts_with($key, 'product_') && $value) {
+                    $orderDetailId = (int) str_replace('product_', '', $key);
+                    $quantityKey = 'quantity_' . $orderDetailId;
 
-                $selectedProducts[] = [
-                    'id_order_detail' => $orderDetailId,
-                    'quantity' => (int) $submittedData[$quantityKey],
-                ];
+                    $selectedProducts[] = [
+                        'id_order_detail' => $orderDetailId,
+                        'quantity' => (int) $submittedData[$quantityKey],
+                    ];
+                }
             }
-        }
 
-        $command = new MergeProductsToShipment(
-            $shipmentId,
-            $targetShipmentId,
-            $selectedProducts
-        );
-        $this->dispatchCommand($command);
+            $command = new MergeProductsToShipment(
+                $shipmentId,
+                $targetShipmentId,
+                $selectedProducts
+            );
+            $this->dispatchCommand($command);
+        } catch (Exception $e) {
+            if ((new ReflectionClass($e))->getShortName() === 'CannotEditShipmentShippedException') {
+                $this->addFlash('error', $this->trans('You cannot split shipment when is already shipped.', [], 'Admin.Orderscustomers.Notification'));
+            }
```

Comment:
You should rethrow the exceptions you don't handle

```suggestion
            } else {
                throw $e;
            }
```